ivy_tests/test_ivy/test_functional/test_core/test_device.py

```diff
@@ -605,13 +606,60 @@ def test_function_unsupported_devices(
     assert sorted(tuple(exp)) == sorted(res)
 
 
-# Still to Add #
-# ---------------#
+def get_gpu_mem_usage(device='gpu:0'):
+    handle = _get_nvml_gpu_handle(device)
+    info = pynvml.nvmlDeviceGetMemoryInfo(handle)
+    return (info.used / info.total) * 100
+
+@handle_test(fn_tree="clear_cached_mem_on_dev")
+def test_clear_cached_mem_on_dev():
+    devices = _get_possible_devices()
+    for device in devices:
+        # Testing on only GPU since clearing cache mem is relevant
+        # for only CUDA devices
+        if 'gpu' in device:
+            arr = ivy.random_normal(shape=(10000,10000), dtype="float32", device=device)
+            del arr
+            before = get_gpu_mem_usage(device)
+            ivy.clear_cached_mem_on_dev(device)
+            after = get_gpu_mem_usage(device)
+            assert before > after
+
+
+def get_cpu_percent():
+    output = subprocess.check_output(['top', '-bn1'])
+    cpu_percent = float(re.search(b'%Cpu\(s\):\s+([\d.]+)\s+us', output).group(1))
+    return cpu_percent
 
 
-# clear_cached_mem_on_dev
-# used_mem_on_dev # working fine for cpu
```

Comment:
Let's add this as a `ToDo` comment to `test_used_mem_on_dev`, just so that we don't forget